### DISCUSSION TITLE
Data selection in IPF

### DISCUSSION THREAD

**Discussion by GitHubUserWithoutAName** [mtex-toolbox/mtex#829] [Ask Anything] [answered]
Dear MTEX community,

I am currently considering the best way to select regions of data points from IPF plots. The most convenient way to achieve what I need would be by something similar to how one can select EBSD data points by spatial coordinates with commands like `selectPolygon` and `inpolygon`, operating on EBSD map plots. Since I do not know of any similar commands that work on IPF or PF plots: Is there a direct way to achieve this that I am missing? 

If not, I will probably have to program some logic of my own with which I can select data points based on their orientation (I am thinking of thinks like `calcCluster` and `findByOrientation(grains,ori,epsilon)`). The downside would be that it seems like I am restricted to using one threshold angle in all crystallographic directions (i.e. being restricted to a sphere in orientation space, if I am not mistaken), whereas manually selecting a polygon would allow me to select arbitrary shapes / clusters. That probably means that I have to manually concatenate multiple clusters, but I'll have to work that out.

Thanks for your help!

**Comment by kilir** [comment, marked as answer]
Hi,
have a look here: https://github.com/mtex-toolbox/mtex/issues/299#issuecomment-361072631
This let's you click a polygon on a sphere and select the points inside the polygon. It should mostly work for Miller as well - maybe with some minor changes.

Cheers,
Rüdiger

**Reply by GitHubUserWithoutAName** [reply]
Edit: Managed to get it to work - updated post to reflect this, removed second question and tagged as solved.

Dear Rüdiger,

thanks a lot for your help. After some playing around, I managed to make this work based on your link. Your function `inpolygonSP` seems to operate just fine on both `Miller` and `vector3d`. On this basis, I managed to select data from both pole figures and inverse pole figures, defining the polygon and checking data points in terms of vector3d for PFs and Miller for IPFs. In the IPF case, one can e.g. use the pos output from  `getDataCursorPos` and cast it to `Miller ` (the same way as is done for tooltip creation in `plotIPDF `for orientation data) in order to obtain the vertices of the polygon in terms of `Miller`.

If you (or anyone else knowledgeable about this) happens to check in on this thread again, I'd still be interested in my original second question: 
 I understand (from the thread you linked) that for the PF case, using `sphericalRegion.byVertices` and `sphericalRegion.checkInside` could be a shortcut. While I think understand in principle how to define a sphericalRegion using N and alpha (normal vector and cosine), how does this work for vector3d vertices? I tried digging through the documentation and implementation but could't work out what is going on. Can I supply an arbitrary number of vertices that make up a polygon on the sphere? I have to supply those vertices in terms of the points on the sphere, right? No matter what I try, I end up with a spherical region with only one edge normal: `NaN, NaN, NaN`, e.g. for `sR = sphericalRegion.byVertices(xvector, yvector, zvector)`. All three elementary vectors are on the surface of the unit sphere (with `r = 1`), right? Or do I completely miss what .byVertices is supposed to do? I am using MTEX 5.4.0, by the way.

Thanks again for your help!
